pep/app/pods/components/search/item/component.ts

```diff
@@ -0,0 +1,81 @@
+import Component from '@glimmer/component';
+import { action, computed } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { PreferenceDocumentsKey, PreferenceKey } from 'pep/constants/preferences';
+import CurrentUserService from 'pep/services/current-user';
+import Document from 'pep/pods/document/model';
+import SidebarService from 'pep/services/sidebar';
+import FastbootService from 'ember-cli-fastboot/services/fastboot';
+
+interface SearchItemArgs {
+    item: Document;
+    openResultPreview: () => void;
+}
+
+export default class SearchItem extends Component<SearchItemArgs> {
+    @service currentUser!: CurrentUserService;
+    @service sidebar!: SidebarService;
+
+    /**
+     * Using a computed here so we A) dont dip into the local storage too often and B) so that this
+     * is recomputed whenever the preferences object is updated
+     *
+     * @readonly
+     * @memberof SearchItem
+     */
+    @computed('currentUser.preferences', 'args.item')
+    get favorited() {
+        return this.currentUser.hasDocument(PreferenceKey.FAVORITES, this.args.item);
```

Comment:
switching to ID-only storage, this would become a simple `array.includes()` check.